src/lib/functions/integrations/sprig.ts

```diff
@@ -25,18 +24,33 @@ export const lazyLoadAndInitSprig = () => {
   return window['Sprig'];
 }
 
-export const triggerForNewUsers = async () => {
-  const user = await fetchUserProfile();
-  if (!user?.createdAt) {
-    return
+const triggerForNewUsers = async (appContext) => {
+  if (!SPRIG_ID) {
+    // do not initialize if SPRIG_ID config was not set
+    return;
   }
 
-  const userAge = Math.round(Date.now() - +new Date(user.createdAt)) / 86400000;
-  const isNewUser = userAge <= 30;
+  appContext.subscribe(({auth: { user }, integrations}) => {
+    if (integrations?.sprig.match(/^disable/i)) {
```

Comment:
Would it make sense to extract this check into a common helper function that can return the disabled status for any integration?

Reply:
Yes, makes sense. I extracted it separately and added some more checks, thanks! 👍